### PR TITLE
Update to work with psc-0.15.7-20230207

### PR DESCRIPTION
**Pull request by flip111**
https://github.com/thomashoneyman/purescript-halogen-portal/issues/5#issuecomment-1424825638

### CODE REVIEW

File: example/Example/Component/Child.purs [hunk 1]
```diff
@@ -1,10 +1,10 @@
 module Example.Component.Child where
 
-import Prelude
 import Data.Maybe (Maybe(..))
 import Halogen as H
 import Halogen.HTML as HH
 import Halogen.HTML.Events as HE
+import Prelude
```
Comment (thomashoneyman) [inline example/Example/Component/Child.purs]: Could you move this back to the top of the import list?

File: example/Example/Component/Parent.purs [hunk 2]
```diff
@@ -1,16 +1,16 @@
...
 import Data.Foldable (traverse_)
 import Data.Maybe (Maybe(..))
-import Data.Symbol (SProxy(..))
 import Effect.Aff.Class (class MonadAff)
 import Effect.Class.Console as Console
 import Example.Component.Child as Child
 import Halogen as H
 import Halogen.HTML as HH
 import Halogen.Portal as Portal
+import Prelude
```
Comment (thomashoneyman) [inline example/Example/Component/Parent.purs]: Same here — I'd prefer the unqualified import go at the top.

File: packages.dhall [hunk 3]
```diff
@@ -1,23 +1,105 @@
...
+          , "node-readline"
+          , "datetime"
+          , "now"
+          ]
+      , repo =
+          "https://github.com/hdgarrood/purescript-benchotron.git"
+      , version =
+          "v7.0.0"
       }
+-------------------------------
```
Comment (thomashoneyman) [inline packages.dhall]: Could you remove this comment, leaving just the `upstream` code?

File: spago.dhall [hunk 4]
```diff
@@ -1,15 +1,33 @@
...
+You can edit this file as you like.
+
+Need help? See the following resources:
+- Spago documentation: https://github.com/purescript/spago
+- Dhall language tour: https://docs.dhall-lang.org/tutorials/Language-Tour.html
+
+When creating a new Spago project, you can use
+`spago init --no-comments` or `spago init -C`
+to generate this file without the comments in this block.
+-}
```
Comment (thomashoneyman) [inline spago.dhall]:
```suggestion
```

File: src/Halogen/Portal.purs [hunk 5]
```diff
@@ -4,27 +4,29 @@
...
 import Effect.Aff.AVar as AVar
 import Effect.Aff.Class (class MonadAff)
+import Effect.Class (liftEffect)
 import Halogen as H
 import Halogen.Aff (awaitBody)
 import Halogen.HTML as HH
+import Halogen.Subscription as HS
 import Halogen.VDom.Driver as VDom
-import Web.HTML (HTMLElement)
+import Prelude
```
Comment (thomashoneyman) [inline src/Halogen/Portal.purs]: Could you move this to the top of the import list?

File: src/Halogen/Portal.purs [hunk 6]
```diff
@@ -50,6 +52,7 @@ type State query input output n
 -- ```
 --
 -- Another option is to use `H.hoist` to lift your component into `ReaderT`.
+newtype NT :: forall k. (k -> Type) -> (k -> Type) -> Type
```
Comment (thomashoneyman) [inline src/Halogen/Portal.purs]: This is meant to be used with monads, so there's no need to be polymorphic in `k`.

```suggestion
newtype NT :: (Type -> Type) -> (Type -> Type) -> Type
```

**Review by thomashoneyman — CHANGES_REQUESTED** [review]

**Comment by thomashoneyman** [comment]
I have some minimal comments, but otherwise :+1: this looks good! Thanks!

**Comment by flip111** [comment]
Reference to nix also has to removed from readme

**Comment by flip111** [comment]
The dependency to `halogen-storybook` is it really necessary to do that way? Everyone using halogen-portal will pull in this dep, no ?

**Comment by thomashoneyman** [comment]
> The dependency to `halogen-storybook` is it really necessary to do that way? Everyone using halogen-portal will pull in this dep, no ?

Yea we should really have a spago.example.dhall that contains dependencies for the examples, and you run the build with spago build -x spago.example.dhall

**Comment by thomashoneyman** [comment]
Once we get the CI failure fixed I'm happy to merge.

```
[1/1 ModuleNotFound] example/Main.purs:10:1

  10  import Halogen.Storybook (Stories, proxy, runStorybook)
      ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  
  Module Halogen.Storybook was not found.
  Make sure the source file exists, and that it has been provided as an input to the compiler.
```

**Comment by flip111** [comment]
Thanks for the merge and help with the review, it's cool !

**Pull request merged**